src/Tasks/Backup/Zip.php

```diff
@@ -126,4 +135,44 @@ public function count(): int
     {
         return $this->fileCount;
     }
+
+    public function encrypt()
+    {
+        $this->loadEncryption();
+
+        if ($this->getEncryption()) {
+            $this->setPassword($this->getEncryption()->getPassword());
+
+            foreach (range(0, $this->zipFile->numFiles - 1) as $i) {
+                $this->zipFile->setEncryptionIndex($i, $this->getEncryption()->getMethod());
+            }
+        }
+    }
+
+    public function getEncryption(): ?Encryption
+    {
+        return $this->encryption;
+    }
+
+    public function loadEncryption()
```

Comment:
Move all of this logic to a new static method `loadFromConfigFile` on the `BackupEncryptionClass` so the `Zip` becomes smaller